flatten_dict/flatten_dict.py

```diff
@@ -41,9 +44,9 @@ def flatten(d, reducer='tuple', inverse=False):
     flat_dict = {}
 
     def _flatten(d, parent=None):
-        for key, value in six.viewitems(d):
+        for key, value in (six.viewitems(d) if isinstance(d, dict) else enumerate(d)):
             flat_key = reducer(parent, key)
-            if isinstance(value, Mapping):
+            if isinstance(value, (Mapping, *enumerate_types)):
```

Comment:
I think this is a new syntax in python 3.5, but we still support 2.7.

Reply:
yeah did quick impl on launch-break which i didn't check results

i made it `tuple([ Mapping ] + enumerate_types)` which is not well looking but works
because of that changed parameter type to list

another way would be `(Mapping, ) + enumerate_types`, did not use it because of extra comma...

if you have another suggestions, please let me know :)

Reply:
Generally, we shouldn't use mutable type (e.g., `list`) as the type of a default argument. See [this](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) for more details.

BTW, we should remove the white space immediately inside the parentheses because PEP8 has that rule, so it should be:
```
(Mapping,) + enumerate_types
```

That comma is quite common in Python. I think most people won't think it's ugly.